modules/library/main/src/test/java/org/geotools/styling/visitor/RenderingSelectorStyleVisitorTest.java

```diff
@@ -0,0 +1,111 @@
+package org.geotools.styling.visitor;
+
+import static org.geotools.styling.visitor.RenderingSelectorStyleVisitor.RENDERING_LEGEND_OPTION;
+import static org.geotools.styling.visitor.RenderingSelectorStyleVisitor.RENDERING_MAP_OPTION;
+import static org.junit.Assert.assertEquals;
+
+import java.util.Arrays;
+import java.util.List;
+import org.geotools.factory.CommonFactoryFinder;
+import org.geotools.styling.FeatureTypeStyle;
+import org.geotools.styling.Rule;
+import org.geotools.styling.Style;
+import org.geotools.styling.StyleBuilder;
+import org.geotools.styling.StyleFactory;
+import org.geotools.styling.Symbolizer;
+import org.junit.Before;
+import org.junit.Test;
+import org.opengis.filter.FilterFactory2;
+
+public class RenderingSelectorStyleVisitorTest {
+
+    StyleBuilder sb;
+    StyleFactory sf;
+    FilterFactory2 ff;
+
+    @Before
+    public void setUp() throws Exception {
+        sf = CommonFactoryFinder.getStyleFactory(null);
+        ff = CommonFactoryFinder.getFilterFactory2(null);
+        sb = new StyleBuilder(sf, ff);
+    }
+
+    @Test
+    public void testMapRenderingSelection() {
+        Style oldStyle = sb.createStyle("FTSName", sf.createPolygonSymbolizer());
+        oldStyle.featureTypeStyles().get(0).getOptions().put(RENDERING_MAP_OPTION, "false");
+
+        Rule rule = sf.createRule();
+        Symbolizer symb1 = sf.createLineSymbolizer(sf.getDefaultStroke(), "geometry");
+        rule.symbolizers().add(symb1);
+        rule.getOptions().put(RENDERING_MAP_OPTION, "false");
+
+        Rule rule2 = sf.createRule();
+        rule2.setName("Rendered rule");
+        Symbolizer symb2 =
+                sf.createPolygonSymbolizer(sf.getDefaultStroke(), sf.getDefaultFill(), "shape");
+        rule2.symbolizers().add(symb2);
+        Symbolizer symb3 =
+                sf.createPolygonSymbolizer(sf.getDefaultStroke(), sf.getDefaultFill(), "extension");
+        symb3.getOptions().put(RENDERING_MAP_OPTION, "true");
+        rule2.symbolizers().add(symb3);
+
+        FeatureTypeStyle fts = sf.createFeatureTypeStyle(rule);
```

Comment:
Each of FTS, Rule and Symbolizer can be in 3 different states:

- Misses the option
- Has the option, it's true
- Has the option, it's ... well, not true (false or other value)

Can you build an example where all 3 states are shown at each level? And then you just pass the option name to the method building the example style (rather than copy&paste), run the visit and run your assertions,

Reply:
fixed